FredBoat/src/main/java/fredboat/command/admin/SetAvatarCommand.java

```
@@ -0,0 +1,78 @@
+package fredboat.command.admin;
+
+import com.mashape.unirest.http.Unirest;
+import com.mashape.unirest.http.exceptions.UnirestException;
+import fredboat.FredBoat;
+import fredboat.commandmeta.abs.Command;
+import fredboat.commandmeta.abs.CommandContext;
+import fredboat.commandmeta.abs.ICommandRestricted;
+import fredboat.messaging.internal.Context;
+import fredboat.perms.PermissionLevel;
+import net.dv8tion.jda.core.entities.Icon;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nonnull;
+import java.io.IOException;
+import java.io.InputStream;
+
+/**
+ * This command allows a bot admin to change the avatar of FredBoat
+ */
+public class SetAvatarCommand extends Command implements ICommandRestricted {
+
+    private static final Logger log = LoggerFactory.getLogger(SetAvatarCommand.class);
+
+    @Override
+    public void onInvoke(CommandContext context) {
+        if (context.args.length > 1) {
+            String imageUrl = context.args[1];
+
+            if (imageUrl.startsWith("http://") || imageUrl.startsWith("https://")) {
+                String type = getImageMimeType(imageUrl);
+
+                if (type.equals("image/jpeg") || type.equals("image/png") || type.equals("image/gif") || type.equals("image/webp")) {
+                    setBotAvatar(context, imageUrl);
+                }
+            }
+        }
```

This silently fails when there are not enough arguments provided (0) or the regex doesn't pick up a link. What should happen instead is that the command help should be posted if users use a command wrong and we notice that.

This is also only solves half of the original issue, as it only accepts links, but not uploaded files.

---

My bad, I will add the Help message now. Also please see my comment below as I'm not sure how the command should accept uploaded files.